ark/phenotyping/som_runner.R

```diff
@@ -2,33 +2,56 @@ library(data.table)
 library(FlowSOM)
 library(rhdf5)
 
+# get the command line arguments
 print("Reading command args")
 args <- commandArgs(trailingOnly=TRUE)
 
-# get path to pixel mat data
-print("Getting pixel mat data path")
-pixelMatPath <- args[1]
+# create the list of fovs
+print("Creating fovs list")
+fovs <- unlist(strsplit(args[1], split=","))
+print(fovs)
 
 # create a vector out of the list of channels provided
 print("Creating marker list")
 markers <- unlist(strsplit(args[2], split=","))
+print(markers)
+
+# get path to pixel mat data
+print("Getting pixel mat data path")
+pixelMatPath <- args[3]
 
-# get the path to the write directory
+print("Getting the pixel subsetted data path")
+pixelSubsetPath <- args[4]
+
+# get the path to the write path
 print("Getting the write path")
-writePath <- args[3]
+pixelWritePath <- args[5]
 
 # read the pixel mat data
-print("Reading the pixel matrix data")
-pixelMatData <- fread(pixelMatPath, select=markers)
+print("Reading the subsetted pixel matrix data for SOM training")
+pixelSubsetData <- fread(pixelSubsetPath, select=markers)
+
+# run the SOM training step
+print("Run the SOM training")
+somResults <- SOM(data=as.matrix(pixelSubsetData))
+
+# using trained SOM, batch cluster the original dataset by fov
+print("Mapping cluster labels")
+
+for (fov in fovs) {
```

Comment:
Printing an update line every 20 FOVs or so would be nice so that users have an estimate of long this is going to take